op-challenger/game/fault/types/position.go

```diff
@@ -36,14 +49,16 @@ func (p Position) TraceIndex(maxDepth int) uint64 {
 	// When we go right, we do a shift left and set the bottom bit to be 1.
 	// To do this in a single step, do all the shifts at once & or in all 1s for the bottom bits.
 	rd := maxDepth - p.depth
-	return uint64(p.indexAtDepth<<rd | ((1 << rd) - 1))
+	rhs := ((1 << rd) - 1)
+	ti := new(big.Int).Or(p.lshIndex(rd), big.NewInt(int64(rhs)))
+	return ti.Uint64()
```

Comment:
This overflows for large depths.

ie this test fails:
```
func TestTraceIndexBigValue(t *testing.T) {
	traceIdx := new(big.Int).Mul(new(big.Int).SetUint64(math.MaxUint64), big.NewInt(4))
	pos := Position{depth: 100, indexAtDepth: traceIdx}
	actual := pos.TraceIndex(100)
	require.Equal(t, traceIdx, actual)
}
```

`TraceIndex` will have to return `big.Int` not `uint64`